tokio/src/signal/windows/sys.rs

```diff
@@ -114,7 +129,15 @@ unsafe extern "system" fn handler(ty: u32) -> BOOL {
     // the handler routine is always invoked in a new thread, thus we don't
     // have the same restrictions as in Unix signal handlers, meaning we can
     // go ahead and perform the broadcast here.
-    if globals.broadcast() {
+    let event_was_handled = globals.broadcast();
+
+    if event_was_handled && event_requires_infinite_sleep_in_handler(ty) {
+        loop {
+            std::thread::park();
+        }
+    }
```

Comment:
Instead of doing a `event_requires_infinite_sleep_in_handler` check a second time here, move this bit of logic inside of `raise_event` (that's the place that already decides whether to spin up a new thread, so it already knows whether it is safe to permanently park the thread there

Reply:
I think you misunderstand; `raise_event` is a test function, this is the actual code.

Reply:
Ah yes this is the test code, apologies! Been a few years since I've looked at this code :sweat_smile: